### PR TITLE
Modify delete using student number

### LINKED ISSUE #55: As a TA, I want to be able to delete students from the class
... so that I can manage students who drop out.

### PR DESCRIPTION
**Pull request by ChangruHenryQian**
Closes #55

### CODE REVIEW

File: src/main/java/seedu/address/logic/Messages.java [hunk 1]
```diff
@@ -14,7 +14,9 @@ public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The peron index provided is invalid";
```
Comment (LWZ19) [inline src/main/java/seedu/address/logic/Messages.java:17]: Small spelling mistake here for 'person'

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -18,29 +17,34 @@ public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes a person. "
+            + "Parameters: "
+            + PREFIX_STUDENTNUMBER + "STUDENT NUMBER"
```
Comment (LWZ19) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:22]: Newline could be added for better readability.  :blush:
```suggestion
    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes a person.\n"
            + "Parameters: "
            + PREFIX_STUDENTNUMBER + "STUDENT NUMBER\n"
```

**Review by LWZ19 — COMMENTED** [review]
LGTM. Just some issues with code style and formatting, and making the code pass CI. :blush:
One small suggestion for future iterations could be to include both delete by index or student number, but both cannot be supplied at once.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -18,29 +17,34 @@ public class DeleteCommand extends Command {
...
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (!this.targetStudentNumber.isValid()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_NUMBER);
+        }
+
+        if (!model.hasPerson(new Person(targetStudentNumber))) {
```
Comment (Cikguseven) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:42]: Is it possible to check if model has person using Student number without creating a Person object?

**Comment by Cikguseven** [comment]
LGTM!

**Pull request merged**